EIPS/eip-4672.md

```diff
@@ -0,0 +1,48 @@
+---
+eip: 4672
+title: address.nonce
+description: A new EVM opcode to get the current nonce of an EOA.
+author: 0xfoobar (@0xfoobar)
+discussions-to: https://ethereum-magicians.org/t/eip-4672-account-nonce-opcode/8171
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-02-01
+requires: none
+---
+
+## Abstract
+
+We propose an additional EVM opcode, `address.nonce` (0x47), which could provide novel protection against sybil attacks and expand the smart contract design space. This would return `0` for an unused externally-owned address (EOA), `0` for a smart contract address, and, for example, `2` for an EOA which has sent 3 transactions. The account.nonce opcode would exactly track the nonce of the last transaction originating from that address.
+
+A full list of EVM opcodes can be tracked here: https://github.com/crytic/evm-opcodes
+
+## Motivation
+
+NFT contracts commonly use checks on `msg.sender` to limit sybil attacks. However, this is easily circumvented by spinning up new addresses, or even nesting address creation within a smart contract. Letting smart contracts check `address.nonce` would add additional costs to this, serving as a “proof of history”. While not bulletproof (as msg.sender is not), it opens up the design space to creative uses.
+
+## Specification
+
+A new opcode is added to the EVM, 0x47, which can be used in Solidity as follows:
```

Comment:
Should probably note the stack arguments instead of using a solidity example. See [EIP-1052](https://eips.ethereum.org/EIPS/eip-1052) for an example.

Reply:
done